lib/ansible/cli/doc.py

```diff
@@ -33,8 +33,32 @@
 display = Display()
 
 
+def _json_encode_fallback(obj):
+    if isinstance(obj, datetime.datetime):
+        return obj.isoformat()
+    else:
+        return to_native(obj)
+
+
+def _json_fix_keys(obj):
+    if isinstance(obj, (Sequence, Set)) and not isinstance(obj, string_types):
+        return [_json_fix_keys(x) for x in obj]
+    elif isinstance(obj, Mapping):
+        result = dict()
+        for k, v in obj.items():
+            result[to_native(k)] = _json_fix_keys(v)
+        return result
+    else:
+        return obj
+
+
+def jsonify(data, **kwargs):
+    data = _json_fix_keys(data)
+    return json.dumps(data, default=_json_encode_fallback, **kwargs)
+
+
 def jdump(text):
-    display.display(json.dumps(text, sort_keys=True, indent=4))
+    display.display(jsonify(text, sort_keys=True, indent=4))
```

Comment:
a simpler fix i was already working on:
`+    display.display(json.dumps(text, cls=AnsibleJSONEncoder, sort_keys=True, indent=4))`

Reply:
That still crashes for non-string keys, i.e. for `ec2_vpc_vpn`. That really seem to need pre-processing to get rid of all non-string keys in mappings.

Reply:
```.py
  File ".../ansible/cli/doc.py", line 59, in jdump
    display.display(json.dumps(text, cls=AnsibleJSONEncoder, sort_keys=True, indent=4))
  File "/usr/lib/python3.8/json/__init__.py", line 234, in dumps
    return cls(
  File "/usr/lib/python3.8/json/encoder.py", line 201, in encode
    chunks = list(chunks)
  File "/usr/lib/python3.8/json/encoder.py", line 431, in _iterencode
    yield from _iterencode_dict(o, _current_indent_level)
  File "/usr/lib/python3.8/json/encoder.py", line 405, in _iterencode_dict
    yield from chunks
  File "/usr/lib/python3.8/json/encoder.py", line 405, in _iterencode_dict
    yield from chunks
  File "/usr/lib/python3.8/json/encoder.py", line 405, in _iterencode_dict
    yield from chunks
  [Previous line repeated 2 more times]
  File "/usr/lib/python3.8/json/encoder.py", line 325, in _iterencode_list
    yield from chunks
  File "/usr/lib/python3.8/json/encoder.py", line 353, in _iterencode_dict
    items = sorted(dct.items())
TypeError: '<' not supported between instances of 'int' and 'str'
```

Reply:
^ that seems to be an expression not really a data type issue, in any case, there is also an existing `jsonify` in module_utils

Reply:
The `jsonify` in this PR is a stripped-down version of the one in module_utils. Anyway, the problem *is* that there are dictionaries with wrong key types in the data. ansible-doc should really not die because of that.